### ISSUE TITLE
AIP original name isn't always in the first dmdSec

### ISSUE THREAD

**Issue by ross-spencer** [artefactual-labs/AIPscan#57]
We need a more sophisticated search mechanism.

Related to https://github.com/archivematica/Issues/issues/1305

**Comment by ross-spencer** [comment]
An example of the complexity here by documenting empty directories:
```xml
  <mets:dmdSec ID="dmdSec_1">
    <mets:mdWrap MDTYPE="PREMIS:OBJECT">
      <mets:xmlData>
        <premis:object xmlns:premis="http://www.loc.gov/premis/v3" xsi:type="premis:intellectualEntity" xsi:schemaLocation="http://www.loc.gov/premis/v3 http://www.loc.gov/standards/premis/v3/premis.xsd" version="3.0">
          <premis:objectIdentifier>
            <premis:objectIdentifierType>UUID</premis:objectIdentifierType>
            <premis:objectIdentifierValue>e944bddb-0cc0-4005-ae30-4a5b3ee84d9f</premis:objectIdentifierValue>
          </premis:objectIdentifier>
          <premis:originalName>%transferDirectory%objects/E1/</premis:originalName>
        </premis:object>
      </mets:xmlData>
    </mets:mdWrap>
  </mets:dmdSec>
  <mets:dmdSec ID="dmdSec_2">
    <mets:mdWrap MDTYPE="PREMIS:OBJECT">
      <mets:xmlData>
        <premis:object xmlns:premis="http://www.loc.gov/premis/v3" xsi:type="premis:intellectualEntity" xsi:schemaLocation="http://www.loc.gov/premis/v3 http://www.loc.gov/standards/premis/v3/premis.xsd" version="3.0">
          <premis:objectIdentifier>
            <premis:objectIdentifierType>UUID</premis:objectIdentifierType>
            <premis:objectIdentifierValue>4a3773b4-1d87-4a4a-aedc-1c1f8fa585fa</premis:objectIdentifierValue>
          </premis:objectIdentifier>
          <premis:originalName>%transferDirectory%objects/0B/E1/</premis:originalName>
        </premis:object>
      </mets:xmlData>
    </mets:mdWrap>
  </mets:dmdSec>
  <mets:dmdSec ID="dmdSec_3">
    <mets:mdWrap MDTYPE="PREMIS:OBJECT">
      <mets:xmlData>
        <premis:object xmlns:premis="http://www.loc.gov/premis/v3" xsi:type="premis:intellectualEntity" xsi:schemaLocation="http://www.loc.gov/premis/v3 http://www.loc.gov/standards/premis/v3/premis.xsd" version="3.0">
          <premis:objectIdentifier>
            <premis:objectIdentifierType>UUID</premis:objectIdentifierType>
            <premis:objectIdentifierValue>e6db3d3e-7396-4439-82d0-e60156ca7ad5</premis:objectIdentifierValue>
          </premis:objectIdentifier>
          <premis:originalName>empty-dirs-e6db3d3e-7396-4439-82d0-e60156ca7ad5</premis:originalName>
        </premis:object>
      </mets:xmlData>
    </mets:mdWrap>
  </mets:dmdSec>
  <mets:dmdSec ID="dmdSec_4">
    <mets:mdWrap MDTYPE="PREMIS:OBJECT">
      <mets:xmlData>
        <premis:object xmlns:premis="http://www.loc.gov/premis/v3" xsi:type="premis:intellectualEntity" xsi:schemaLocation="http://www.loc.gov/premis/v3 http://www.loc.gov/standards/premis/v3/premis.xsd" version="3.0">
          <premis:objectIdentifier>
            <premis:objectIdentifierType>UUID</premis:objectIdentifierType>
            <premis:objectIdentifierValue>42498a16-7315-4302-ada1-51113517b923</premis:objectIdentifierValue>
          </premis:objectIdentifier>
          <premis:originalName>%transferDirectory%objects/0A/</premis:originalName>
        </premis:object>
      </mets:xmlData>
    </mets:mdWrap>
  </mets:dmdSec>
  <mets:dmdSec ID="dmdSec_5">
    <mets:mdWrap MDTYPE="PREMIS:OBJECT">
      <mets:xmlData>
        <premis:object xmlns:premis="http://www.loc.gov/premis/v3" xsi:type="premis:intellectualEntity" xsi:schemaLocation="http://www.loc.gov/premis/v3 http://www.loc.gov/standards/premis/v3/premis.xsd" version="3.0">
          <premis:objectIdentifier>
            <premis:objectIdentifierType>UUID</premis:objectIdentifierType>
            <premis:objectIdentifierValue>517a69ee-322a-4f93-a991-c38e78846f13</premis:objectIdentifierValue>
          </premis:objectIdentifier>
          <premis:originalName>%transferDirectory%objects/0B/</premis:originalName>
        </premis:object>
      </mets:xmlData>
    </mets:mdWrap>
  </mets:dmdSec>
```
And Dataverse:
```xml
  <mets:dmdSec ID="dmdSec_1" CREATED="2020-09-15T12:19:50" STATUS="original">
    <mets:mdWrap MDTYPE="DDI">
      <mets:xmlData>
        <ddi:codebook xmlns:ddi="http://www.icpsr.umich.edu/DDI" version="2.5" xsi:schemaLocation="http://www.ddi:codebook:2_5 http://www.ddialliance.org/Specification/DDI-Codebook/2.5/XMLSchema/codebook.xsd">
          <ddi:stdyDscr>
            <ddi:citation>
              <ddi:titlStmt>
                <ddi:titl>A study of my afternoon snacks</ddi:titl>
                <ddi:IDNo agency="doi">https://doi.org/10.5072/FK2/QAWS8O</ddi:IDNo>
              </ddi:titlStmt>
              <ddi:rspStmt>
                <ddi:AuthEnty affiliation="Channel 4">Noel Fielding</ddi:AuthEnty>
                <ddi:AuthEnty affiliation="BBC">Mary Berry</ddi:AuthEnty>
                <ddi:AuthEnty affiliation="Freelance">Hollywood, Paul</ddi:AuthEnty>
              </ddi:rspStmt>
              <ddi:distStmt>
                <ddi:distrbtr>Scholars Portal Dataverse</ddi:distrbtr>
              </ddi:distStmt>
              <ddi:verStmt>
                <ddi:version date="2019-03-28T15:28:07Z" type="RELEASED">6.0</ddi:version>
              </ddi:verStmt>
            </ddi:citation>
            <ddi:dataAccs>
              <ddi:useStmt>
                <ddi:restrctn>CC0 Waiver</ddi:restrctn>
              </ddi:useStmt>
            </ddi:dataAccs>
          </ddi:stdyDscr>
        </ddi:codebook>
      </mets:xmlData>
    </mets:mdWrap>
  </mets:dmdSec>
  <mets:dmdSec ID="dmdSec_2" CREATED="2020-09-15T12:19:50" STATUS="original">
    <mets:mdRef LABEL="dataset.json" xlink:href="metadata/dataset.json" MDTYPE="OTHER" LOCTYPE="OTHER" OTHERLOCTYPE="SYSTEM"/>
  </mets:dmdSec>
  <mets:dmdSec ID="dmdSec_1">
    <mets:mdWrap MDTYPE="PREMIS:OBJECT">
      <mets:xmlData>
        <premis:object xmlns:premis="http://www.loc.gov/premis/v3" xsi:type="premis:intellectualEntity" xsi:schemaLocation="http://www.loc.gov/premis/v3 http://www.loc.gov/standards/premis/v3/premis.xsd" version="3.0">
          <premis:objectIdentifier>
            <premis:objectIdentifierType>UUID</premis:objectIdentifierType>
            <premis:objectIdentifierValue>e399f3de-344f-4c63-8361-cd6e3b0093f8</premis:objectIdentifierValue>
          </premis:objectIdentifier>
          <premis:originalName>snacks-e399f3de-344f-4c63-8361-cd6e3b0093f8</premis:originalName>
        </premis:object>
      </mets:xmlData>
    </mets:mdWrap>
  </mets:dmdSec>
```

**Comment by peterVG** [comment]
Can we close this issue after PR https://github.com/artefactual-labs/AIPscan/pull/59? 

We still have an Epic for upstream METS-RW issues here: Issue https://github.com/artefactual-labs/AIPscan/issues/53
